src/tracer.js

```diff
@@ -253,6 +273,38 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling if the parent span isn't sampled
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
```

Comment:
If we were to upsample this span but later change the operation name, we will attempt to sample again. If  we decide to not sample, the upsamplingDecision on the parent is still set to true and the next child span of the parent will be sampled. Is that ok?